java/src/com/swiftnav/sbp/client/SBPForwarder.java

```diff
@@ -0,0 +1,34 @@
+package com.swiftnav.sbp.client;
+
+import com.swiftnav.sbp.SBPMessage;
+import com.swiftnav.sbp.observation.MsgEphemeris;
+
+import java.io.IOException;
+
+public class SBPForwarder extends Thread {
+    private Iterable<SBPMessage> source;
+    private SBPSender sink;
+
+    public SBPForwarder(Iterable<SBPMessage> source_, SBPSender sink_) {
+        source = source_;
+        sink = sink_;
+    }
+
+    @Override
+    public void run() {
+        for (SBPMessage msg : source) {
+            System.out.println(msg.toString());
+            try {
+                if (msg.type == MsgEphemeris.TYPE) {
```

Comment:
Also, where are these messages being forwarded to? I seem to remember that the Python client forwarded all messages to the logger and that was it.